contrib/automation_tests/test_cases/capture_window.py

```diff
@@ -188,9 +188,22 @@ def _show_capture_window(self):
         logging.info('Showing capture window')
         self.find_control("TabItem", "Capture").click_input()
 
-    def _execute(self, length_in_seconds: int = 5):
-        self._show_capture_window()
+    def _set_capture_options(self, collect_thread_states: bool):
+        capture_tab = self.find_control('Group', "CaptureTab")
+
+        logging.info('Opening "Capture Options" dialog')
+        capture_options_button = self.find_control('Button', 'Capture Options', parent=capture_tab)
+        capture_options_button.click_input()
+
+        collect_thread_states_checkbox = self.find_control('CheckBox', 'Collect thread states')
```

Comment:
I'm a bit suprised this is working TBH - is the dialog that is being opened parented under the main window? Because by default, the current top window is cached (see `orbit_e2e.E2ETestSuite.top_window`) for performance reasons, so when a new top window appears, this cache needs to be refreshed explicitly.

Could you double-check (in accessibility insights e.g.) if the dialog is located underneath the current main window? Just want to make sure there is no error in the caching and / or this doesn't just work "by accident".

Reply:
From Accessibility Insights the dialog is indeed underneath the main window. What do you think the best approach is? Probably passing the dialog window to `find_control` as per the other comment?

Reply:
Interesting, because context-menus e.g. show up as new top level window. Probably because the Dialog is modal...
I think the most stable way is to find the dialog using `find_control` without explicit parent once, and then passing it explicitly as parent to the remaining `find_control` calls.

Reply:
Yep, done like that, please have another look.